addons/binding/org.openhab.binding.kodi/src/main/java/org/openhab/binding/kodi/internal/KodiAudioSink.java

```diff
@@ -69,15 +70,18 @@ public void process(AudioStream audioStream) throws UnsupportedAudioFormatExcept
             // it is an external URL, the speaker can access it itself and play it.
             URLAudioStream urlAudioStream = (URLAudioStream) audioStream;
             url = urlAudioStream.getURL();
-        } else {
+        } else if (audioStream instanceof FixedLengthAudioStream) {
+            FixedLengthAudioStream fixedLengthAudioStream = (FixedLengthAudioStream) audioStream;
             if (callbackUrl != null) {
-                // we serve it on our own HTTP server
-                String relativeUrl = audioHTTPServer.serve(audioStream);
+                // we serve it on our own HTTP server for 30 seconds as Kodi requests the stream several times
+                String relativeUrl = audioHTTPServer.serve(fixedLengthAudioStream, 30);
                 url = callbackUrl + relativeUrl;
             } else {
                 logger.warn("We do not have any callback url, so kodi cannot play the audio stream!");
                 return;
             }
+        } else {
+            throw new UnsupportedAudioFormatException("Kodi can only handle FixedLengthAudioStreams.", null);
```

Comment:
@cweitkamp AllPlay uses the exception internally and logs a warning if it is thrown, so similar to my first approach. AllPlay uses only `org.eclipse.smarthome.core.audio.AudioHTTPServer.serve(AudioStream)` because it obviously has not the problem like Kodi and Chromecast to access the file multiple times.

@kaikreuzer I had a deeper look into the requests, Kodi does 3 HEAD requests first and then 3 GET requests on the resource. Better than 6 but doesn't help :-/
Kodi can stream fixed length audio and URL audio streams, perhaps also `JavaSoundInputStream` (the only remaining implementation of `AudioStream`), but as I said I don't know how to test. It does not seem to be used in the code either.